### ISSUE TITLE
Feature Request: Ability to select PTZ presets

### ISSUE THREAD

**Issue by julianrinaldi** [briis/securityspy#25]
It would be great to be able to trigger PTZ position presets through this integration.

**Comment by briis** [comment]
I will look at it.

**Comment by briis** [comment]
I will make this request. But I will only implement to ability to activate a PTZ Presets, not the other PTZ Capabilities, as I don't want to replicate the whole SecuritySpy App.

I have not PTZ cameras, so I will rely on you to test this.

**Comment by briis** [comment]
As I don't have any PTZ cameras, I am not exactly sure how this works. Where are the Presets defined? In the SecuritySpy or on the Cameras? How do you see what presets are defined? Could you make a few screenshots?

**Comment by briis** [comment]
Could I also ask you to run this command, replacing the text in CAPITAL letters with your SecuritySpy info and post the result here: `http://USERNAME:PASSWORD@IP_ADDRESS:PORT/systemInfo`

Before you post, **please make sure to remove any data about authentication from the data**. I am basically just interrested to see what comes back, as this data is used to build a lot of the sensors. If it is as I think, I might be able to create a Select entity, from where you can select a preset very easy from the UI.

**Comment by julianrinaldi** [comment]
```
<system>
<server>
<name>SecuritySpy</name>
<version>5.4.0b1</version>
<uuid>UUID</uuid>
<ddns-name>DDNS</ddns-name>
<wan-address>WAN</wan-address>
<server-name>SecuritySpy</server-name>
<bonjour-name/>
<ip1>IP</ip1>
<ip2>IP</ip2>
<http-enabled>yes</http-enabled>
<https-enabled>yes</https-enabled>
<http-port>1459</http-port>
<https-port>8001</https-port>
<http-port-wan>80</http-port-wan>
<https-port-wan>8001</https-port-wan>
<current-local-time>2021-12-17T15:01:22-05:00</current-local-time>
<seconds-from-gmt>-18000</seconds-from-gmt>
<date-format>MM/DD/YYYY</date-format>
<time-format>12</time-format>
<cpu-usage>19</cpu-usage>
<memory-pressure>0</memory-pressure>
<camera-count>26</camera-count>
</server>
<cameralist>
<camera>
<number>5</number>
<connected>yes</connected>
<width>1920</width>
<height>1080</height>
<mode>active</mode>
<mode-c>armed</mode-c>
<mode-m>armed</mode-m>
<mode-a>armed</mode-a>
<hasaudio>no</hasaudio>
<ptzcapabilities>31</ptzcapabilities>
<mdsensitivity>50</mdsensitivity>
<timesincelastframe>0</timesincelastframe>
<timesincelastmotion>233</timesincelastmotion>
<devicename>ONVIF</devicename>
<devicetype>Network</devicetype>
<address>IP</address>
<port>0</port>
<port-rtsp>0</port-rtsp>
<request/>
<name>Driveway</name>
<overlay>no</overlay>
<overlaytext>+n\n+d</overlaytext>
<transformation>0</transformation>
<audio_network>no</audio_network>
<audio_devicename/>
<md_enabled>yes</md_enabled>
<md_sensitivity>50</md_sensitivity>
<md_triggertime_x2>0</md_triggertime_x2>
<md_capture>yes</md_capture>
<md_capturefps>0</md_capturefps>
<md_precapture>0</md_precapture>
<md_postcapture>15</md_postcapture>
<md_captureimages>no</md_captureimages>
<md_uploadimages>no</md_uploadimages>
<md_recordaudio>no</md_recordaudio>
<md_audiotrigger>no</md_audiotrigger>
<md_audiothreshold>50</md_audiothreshold>
<action_scriptname>Web-i Activate Relay 1.scpt</action_scriptname>
<action_soundname/>
<action_resettime>5</action_resettime>
<tl_capture>yes</tl_capture>
<tl_recordaudio>no</tl_recordaudio>
<current-fps>29.851</current-fps>
<video-format>H.264</video-format>
<audio-format/>
<schedule-id-cc>1</schedule-id-cc>
<schedule-id-mc>1</schedule-id-mc>
<schedule-id-a>1</schedule-id-a>
<schedule-override-cc>0</schedule-override-cc>
<schedule-override-mc>0</schedule-override-mc>
<schedule-override-a>0</schedule-override-a>
<preset-name-1>Main</preset-name-1>
<preset-name-2>Parking Spot</preset-name-2>
<preset-name-3>Wildey</preset-name-3>
<preset-name-4>1004 N 3rd</preset-name-4>
<preset-name-5>Front Vestibule</preset-name-5>
<preset-name-6>Parking</preset-name-6>
<preset-name-7>Full Left</preset-name-7>
<preset-name-8>Full Right</preset-name-8>
<preset-name-9/>
<preset-name-10/>
<permissions>65023</permissions>
<capture-path>/Volumes/Security Cameras B/Driveway</capture-path>
</camera>
```

Cameras that are not PTA show <ptzcapabilities> as 0.

**Comment by briis** [comment]
Super, this is really helpful. I will try and build something based on this.

**Comment by briis** [comment]
From the data, there is no way to tell what PTZ preset is currently active, so I suggest, that instead of a Select entity, I use the new `button` entity, available as of Home Assistant 2021.12. This is basically an intermediate switch, that just activates a function but does not track state. I could then create a button for each preset defined for each Camera, and all you have to do, is press that button in th UI or activate the service `button.press` from an automation.

Would that work for you?

**Comment by briis** [comment]
I had to realise this is hard to develop without a PTZ Camera so I can test what is going on. So I just ordered a cheap one, and should get it before the holidays. When I have it, I will start working on this.

**Comment by julianrinaldi** [comment]
> Would that work for you?

Sure, that works. I was also wondering if making a "remote" entity might be feasible as an easy way to control PTZ instead of just using the presets. But that's probably not worth it. I have been using your great integration so long I forgot that Home Assistant has had ONVIF PTZ support built-in for a while. So I can actually trigger presets through that. And I just looked at the SecuritySpy server specs, which I should have done a while ago, and I didn't realize how easily presets can be triggered from just a simple URL. Not sure why I didn't think of that originally. Anyway, I really appreciate your work on this, and having buttons directly in the integration would still be nice.

**Comment by briis** [comment]
I have now released a Beta of V1.1.0. This contains support for PTZ. The cheap Camera I bought only partially supports ONVIF PTZ, so I have not been able to test if activating a Preset works. 
If you installed through HACS, you must activate *Beta Releases* to see it.

**Comment by julianrinaldi** [comment]
Works perfectly. Thanks for implementing this so quickly. Truly impressed. I wish you had one of those buy me a coffee links so I could send something your way.

**Comment by julianrinaldi** [comment]
I also just realized you're the developer behind the Unifi Protect integration which is amazing. Thanks for that as well.

**Comment by briis** [comment]
Your thank you is enough  👍🏻 -  I get a lot from the Community in Home Assistant, so just glad to be able to give something back.

And yes I am also behind UniFi Protect, I started that project back when I started using Home Assistant. As of the last few month I got a lot of help from much more experienced people than myself on UniFi Protect, so that is really taking off now - and hopefully we will see it as part of Home Assistant core in February or March 2022.

Glad the PTZ stuff works, and let me know if there is something else that could be useful. (I am not really using SecuritySpy day-to-day, as most of my Cameras are on Protect)

**Comment by julianrinaldi** [comment]
My only other "wish" is that the online status of the cameras wasn't an attribute of the camera video feed itself. I have a few cheap Chinese knockoff cameras that sometimes disconnect from wifi, and they can be fixed with power cycling the smart plug they're plugged into. But I don't have any need to have the camera feeds for those specific cameras in Home Assistant. I actually only need to display 2-3 of the camera feeds of the 20+ cameras I have in Home Assistant. But having the ability to see their "online" status is useful for me. So ideally I would have the actual camera video feed entities of most of my cameras disabled in Home Assistant while also being able to see their online status.

**Comment by briis** [comment]
I can create a binary_sensor that is on or off depending on online status. Would you mind creating a new issue for that, then I will close this.

**Comment by julianrinaldi** [comment]
I just realized there aren't buttons for zooming in/out. Would it be possible to add these?

**Comment by briis** [comment]
I will add these.

**Comment by briis** [comment]
I just released 1.1.1 where this is now included.

**Comment by julianrinaldi** [comment]
Awesome. Works great. Thanks.
